network/p2p/blob/blob_service.go

```diff
@@ -109,26 +119,35 @@ func NewBlobService(
 	opts ...network.BlobServiceOption,
 ) (*blobService, error) {
 	bsNetwork := bsnet.NewFromIpfsHost(host, r, bsnet.Prefix(protocol.ID(prefix)))
-	blockStore, err := blockstore.CachedBlockstore(
-		context.Background(),
-		blockstore.NewBlockstore(ds),
-		blockstore.DefaultCacheOpts(),
-	)
-	if err != nil {
-		return nil, fmt.Errorf("failed to create cached blockstore: %w", err)
-	}
+
+	blockStore := blockstore.NewBlockstore(ds)
+
 	bs := &blobService{
 		prefix: prefix,
 		config: &BlobServiceConfig{
 			ReprovideInterval: DefaultReprovideInterval,
+			SkipBloomCache:    false, // default: use cached blockstore
 		},
 		blockStore: blockStore,
 	}
 
+	// Apply options before creating blockstore, as SkipBloomCache affects blockstore creation
 	for _, opt := range opts {
 		opt(bs)
 	}
 
+	if bs.config.SkipBloomCache {
+		cachedBlockStore, err := blockstore.CachedBlockstore(
+			context.Background(),
+			blockStore,
+			blockstore.DefaultCacheOpts(),
+		)
+		if err != nil {
+			return nil, fmt.Errorf("failed to create cached blockstore: %w", err)
+		}
+		bs.blockStore = cachedBlockStore
+	}
```

Comment:
The condition appears to be inverted.  We should check `!SkipBloomCache` before creating `cachedBlockStore`.

```suggestion
	if !bs.config.SkipBloomCache {
		cachedBlockStore, err := blockstore.CachedBlockstore(
			context.Background(),
			blockStore,
			blockstore.DefaultCacheOpts(),
		)
		if err != nil {
			return nil, fmt.Errorf("failed to create cached blockstore: %w", err)
		}
		bs.blockStore = cachedBlockStore
	}
```